javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -641,11 +640,6 @@ public T head() {
     }
 
     @Override
-    public Option<T> headOption() {
```

Comment:
this doesn't return `M`, therefore it can be implemented in the parent, where which has `T` already.
Note: reducing the number of method decreased the code coverage slightly.